### PR TITLE
More customizable confirm

### PR DESCRIPTION
**Pull request by Alex1990**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by Alex1990** [comment]
Fix the issue #6722.

Should we need to add a link to [rc-dialog API] in the Modal document?

File: components/modal/confirm.tsx [hunk 1]
```diff
@@ -76,18 +71,34 @@ export default function confirm(config) {
     [`${prefixCls}-${props.type}`]: true,
   }, props.className);
 
+  const dialogProps = assign({
```
Comment (benjycui) [inline components/modal/confirm.tsx]: This code style is weird...

Just follow the old style. Just pass `zIndex` to `rc-dialog` is enough..

Comment (Alex1990) [inline components/modal/confirm.tsx]: Sorry. What about this?

```js
const defaultProps = {
  visible: true,
  title: '',
  footer: '',
  transitionName: 'zoom',
  maskTransitionName: 'fade',
  maskClosable: false, // 默认为 false，保持旧版默认行为
  style: {},
  width: 416,
};
const dialogProps = omit(props, [
  'type',
  'prefixCls',
  'className',
  'iconType',
  'title',
  'content',
  'okCancel',
  'okText',
  'cancelText',
  'onOk',
  'onCancel',
]);

ReactDOM.render(
  <Dialog
    {...defaultProps}
    {...dialogProps}
```

And, what exactly is the "old style"?

I see an old issue https://github.com/ant-design/ant-design/issues/4488. If you have resolved that problem completely, these issue and pr are not created. So, our time is saved.

Comment (afc163) [inline components/modal/confirm.tsx]: Just add `zIndex`, no need to refactor code like this.

Comment (Alex1990) [inline components/modal/confirm.tsx]: Okay! You are the boss.

**Pull request closed without merging**